onchange_helper/models/ir_rule.py

```diff
@@ -26,7 +26,10 @@ def play_onchanges(self, values, onchange_fields):
     all_values = values.copy()
     for field in self._fields:
         if field not in all_values:
-            all_values[field] = False
+            # If self is a record (play onchange on existing record)
+            # we take the value of the field
+            # If self is an empty record we will have an empty value
+            all_values[field] = self[field]
```

Comment:
you should use convert to write to transform the record's properties into the format expected by 'new'